### PR TITLE
add `py.typed`

### PR DESCRIPTION
**Pull request by jorenham**
Because according to the [official typing spec](https://typing.readthedocs.io/en/latest/spec/distributing.html#packaging-type-information):

> For namespace packages (see [PEP 420](https://peps.python.org/pep-0420/)), the `py.typed` file should be in the submodules of the namespace, to avoid conflicts and for clarity.

Which makes sense in this case, as this project looks all nice and type-hinted 👌🏻

### CODE REVIEW

**Comment by csala** [comment]
@jorenham I just saw that codecov is making the workflow fail. I opened issue #7 to try to fix it.

But apart from this, [`pre-commit` is failing due to end-of-file character](https://github.com/csala/mdformat-pyproject/actions/runs/12180940412/job/33978123501). Would you mind sorting it out?

**Comment by jorenham** [comment]
Hmm I thought the `py.typed` was required to be literally empty 🤔  But I'll add the newline anyway, as static type-checkers probably won't care about it anyway 👌🏻

**Comment by jorenham** [comment]
> Hmm I thought the `py.typed` was required to be literally empty 🤔 But I'll add the newline anyway, as static type-checkers probably won't care about it anyway 👌🏻

Oh huh apparently I misunderstood; pre-commit actually want to remove it 🤔

File: mdformat_pyproject/py.typed [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (jorenham) [inline mdformat_pyproject/py.typed]:
```suggestion
```

**Comment by jorenham** [comment]
GitHub's web editor messed it up again... 
It should be fixed now.

**Comment by csala** [comment]
Thanks @jorenham Yes. The github editor is quick, but frequently not the best option :sweat_smile: 

I now fixed the codecov problem. Would you mind merging the main branch into yours, to fix it here?

**Comment by csala** [comment]
Screw it. Now codecov action fails because the token secret is not shared with an external PR.
I'm merging this anyway, since we know that it everything is alright, and I'll figure out the codecov thing for contributions separately.

**Pull request merged**